cpp/CMakeLists.txt

```diff
@@ -472,7 +472,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARROW_CXXFLAGS}")
 
 # For any C code, use the same flags. These flags don't contain
 # C++ specific flags.
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARROW_CXX_FLAGS} ${CXX_COMMON_FLAGS}")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARROW_CXXFLAGS} ${CXX_COMMON_FLAGS}")
```

Comment:
Good catch!
Could you use `${CXX_COMMON_FLAGS} ${ARROW_CXXFLAGS}` order to allow overriding `${CXX_COMMON_FLAGS}` by `${ARROW_CXXFLAGS}`?

Reply:
Thank you for your comment! Changed the order in 6e7aaa3.